tests/linalg_test.py

```diff
@@ -716,6 +716,17 @@ def testPinv(self, shape, dtype, rng_factory):
     self._CheckAgainstNumpy(onp.linalg.pinv, np.linalg.pinv, args_maker,
                             check_dtypes=True, tol=1e-3)
     self._CompileAndCheck(np.linalg.pinv, args_maker, check_dtypes=True)
+    # TODO(phawkins): 1e-1 seems like a very loose tolerance.
+    jtu.check_grads(np.linalg.pinv, args_maker(), 2, rtol=1e-1)
+
+
+  def testPinvGradIssue2792(self):
+    def f(p):
+      a = np.array([[0., 0.],[-p, 1.]], np.float32) * 1 / (1 + p**2)
```

Comment:
Another scenario where I think the current gradient implementation would fail is any case where there are repeated singular values in the SVD, e.g., `np.eye(2)`.

Reply:
Added `np.eye(2)` as a test case. This passes under the new gradient but failed under both variants of differentiating through the SVD.